playbooks/common/openshift-cluster/upgrades/post_control_plane.yml

```diff
@@ -9,77 +9,94 @@
-    registry_url: "{{ openshift.master.registry_url }}"
-    openshift_examples_import_command: replace
-  - role: openshift_hosted_templates
-    registry_url: "{{ openshift.master.registry_url }}"
-    openshift_hosted_templates_import_command: replace
-  pre_tasks:
 
+  pre_tasks:
+  - name: Load lib_openshift modules
+    include_role:
+      name: lib_openshift
   # TODO: remove temp_skip_router_registry_upgrade variable. This is a short term hack
   # to allow ops to use this control plane upgrade, without triggering router/registry
   # upgrade which has not yet been synced with their process.
   - name: Collect all routers
-    command: >
-      {{ oc_cmd }} get pods --all-namespaces -l 'router' -o json
+    oc_obj:
+      state: list
+      kind: pods
+      all_namespaces: True
+      selector: 'router'
     register: all_routers
-    failed_when: false
-    changed_when: false
     when: temp_skip_router_registry_upgrade is not defined
 
-  - set_fact: haproxy_routers="{{ (all_routers.stdout | from_json)['items'] | oo_pods_match_component(openshift_deployment_type, 'haproxy-router') | oo_select_keys_from_list(['metadata']) }}"
-    when: all_routers.rc == 0 and temp_skip_router_registry_upgrade is not defined
+  - set_fact: haproxy_routers="{{ (all_routers.reults.results[0]['items'] | oo_pods_match_component(openshift_deployment_type, 'haproxy-router') | oo_select_keys_from_list(['metadata']) }}"
+    when:
+    - all_routers.results.returncode == 0
+    - temp_skip_router_registry_upgrade is not defined
 
   - set_fact: haproxy_routers=[]
-    when: all_routers.rc != 0 and temp_skip_router_registry_upgrade is not defined
+    when:
+    - all_routers.results.returncode != 0
+    - temp_skip_router_registry_upgrade is not defined
 
   - name: Update router image to current version
-    when: all_routers.rc == 0 and temp_skip_router_registry_upgrade is not defined
+    when:
+    - all_routers.results.returncode == 0
+    - temp_skip_router_registry_upgrade is not defined
     command: >
       {{ oc_cmd }} patch dc/{{ item['labels']['deploymentconfig'] }} -n {{ item['namespace'] }} -p
       '{"spec":{"template":{"spec":{"containers":[{"name":"router","image":"{{ router_image }}","livenessProbe":{"tcpSocket":null,"httpGet":{"path": "/healthz", "port": 1936, "host": "localhost", "scheme": "HTTP"},"initialDelaySeconds":10,"timeoutSeconds":1}}]}}}}'
       --api-version=v1
     with_items: "{{ haproxy_routers }}"
 
   - name: Check for default registry
-    command: >
-      {{ oc_cmd }} get -n default dc/docker-registry
+    oc_obj:
+      state: list
+      kind: dc
+      name: docker-registry
     register: _default_registry
-    failed_when: false
-    changed_when: false
     when: temp_skip_router_registry_upgrade is not defined
 
   - name: Update registry image to current version
-    when: _default_registry.rc == 0 and temp_skip_router_registry_upgrade is not defined
+    when:
+    - _default_registry.rc == 0
```

Comment:
Double checking ... does the ``obj`` module pass the ``rc`` through? I see direct references to ``returncode`` but not ``rc``.